source/common/network/listener_impl.cc

```diff
@@ -43,85 +41,92 @@ bool ListenerImpl::rejectCxOverGlobalLimit() {
   return AcceptedSocketImpl::acceptedSocketCount() >= global_cx_limit;
 }
 
-void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* remote_addr,
-                                  int remote_addr_len, void* arg) {
-  ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
-
-  // Wrap raw socket fd in IoHandle.
-  IoHandlePtr io_handle = SocketInterfaceSingleton::get().socket(fd);
-
-  if (rejectCxOverGlobalLimit()) {
-    // The global connection limit has been reached.
-    io_handle->close();
-    listener->cb_.onReject();
-    return;
+void ListenerImpl::onSocketEvent(short flags) {
+  ASSERT(flags & (Event::FileReadyType::Read));
+
+  while (1) {
+    if (!socket_->ioHandle().isOpen()) {
+      PANIC(fmt::format("listener accept failure: {}", errorDetails(errno)));
+    }
+
+    sockaddr_storage remote_addr;
+    socklen_t remote_addr_len = sizeof(remote_addr);
+
+    IoHandlePtr io_handle = socket_->ioHandle().accept(reinterpret_cast<sockaddr*>(&remote_addr),
+                                                       &remote_addr_len, ENVOY_SOCK_NONBLOCK);
```

Comment:
ENVOY_SOCK_NONBLOCK seems to be hard coded.

Do we need the flags argument?  Should the implementations set non-block unconditionally on their own?

Reply:
ENVOY_SOCK_NONBLOCK is just an os independent flag that we can pass to accept since SOCK_NONBLOCK does not exist on windows and darwin. The traditional POSIX accept syscall, on linux at least, does not inherit the O_NONBLOCK flag (see SOCK_NONBLOCK [here](https://man7.org/linux/man-pages/man2/accept.2.html)).

Reply:
The question is if you should hard code use of SOCK_NONBLOCK in OsSysCallsImpl::accept and drop the flags argument to the accept interface method.

Reply:
Apologies, I thought you meant the underlying syscalls, not our accept implementations. I guess we can do that and then if at a later point the need for flags arises, we can re-add the parameter. Reasonable?